src/file_pipe_log/log_file.rs

```diff
@@ -110,30 +125,62 @@ impl<F: FileSystem> LogFileWriter<F> {
     }
 }
 
+/// Build a file reader.
+///
+/// Attention please, the reader do not need a specified `[LogFileFormat]` from
+/// users.
+///
+/// * `[handle]`: standard handle of a log file.
+/// * `[version]`: if `[None]`, reloads the log file header and parse
+/// the relevant `Version` before building the `reader`.
 pub(super) fn build_file_reader<F: FileSystem>(
     system: &F,
     handle: Arc<F::Handle>,
+    version: Option<Version>,
 ) -> Result<LogFileReader<F>> {
     let reader = system.new_reader(handle.clone())?;
-    LogFileReader::open(handle, reader)
+    LogFileReader::open(handle, reader, version)
 }
 
 /// Random-access reader for log file.
 pub struct LogFileReader<F: FileSystem> {
+    pub header: LogFileFormat,
     handle: Arc<F::Handle>,
     reader: F::Reader,
 
     offset: u64,
 }
 
 impl<F: FileSystem> LogFileReader<F> {
-    fn open(handle: Arc<F::Handle>, reader: F::Reader) -> Result<Self> {
-        Ok(Self {
-            handle,
-            reader,
-            // Set to an invalid offset to force a reseek at first read.
-            offset: u64::MAX,
-        })
+    fn open(handle: Arc<F::Handle>, reader: F::Reader, version: Option<Version>) -> Result<Self> {
+        match version {
+            Some(ver) => Ok(Self {
+                header: LogFileFormat::from_version(ver),
+                handle,
+                reader,
+                // Set to an invalid offset to force a reseek at first read.
+                offset: u64::MAX,
+            }),
+            None => {
+                // Here, the caller expected that the given `handle` has pointed to
+                // a log file with valid format. Otherwise, it should return with
+                // `Err`.
+                if handle.file_size()? == 0 {
```

Comment:
This is already handled in `parse_format`.

Reply:
Yes, missed it. It's a duplicated fault.